runtime/capsules/src/mctp/driver.rs

```diff
@@ -0,0 +1,355 @@
+// Licensed under the Apache-2.0 license
+
+use crate::mctp::base_protocol::*;
+use crate::mctp::recv::MCTPRxClient;
+use crate::mctp::send::{MCTPSender, MCTPTxClient};
+
+use core::cell::Cell;
+use core::fmt::Write;
+use romtime::println;
+
+use kernel::grant::{AllowRoCount, AllowRwCount, Grant, UpcallCount};
+use kernel::processbuffer::{ReadableProcessBuffer, WriteableProcessBuffer};
+use kernel::syscall::{CommandReturn, SyscallDriver};
+use kernel::utilities::cells::MapCell;
+use kernel::utilities::leasable_buffer::SubSliceMut;
+use kernel::{ErrorCode, ProcessId};
+
+pub const MCTP_MAX_MESSAGE_SIZE: usize = 4098;
+
+pub const MCTP_SPDM_DRIVER_NUM: usize = 0xA0000;
+pub const MCTP_PLDM_DRIVER_NUM: usize = 0xA0001;
+pub const MCTP_VENDOR_DEFINED_PCI_DRIVER_NUM: usize = 0xA0002;
+
+/// IDs for subscribe calls
+mod upcall {
+    /// Callback for when the message is received
+    pub const MESSAGE_RECEIVED: usize = 0;
+
+    /// Callback for when the message is transmitted.
+    pub const MESSAGE_TRANSMITTED: usize = 1;
+
+    /// Number of upcalls
+    pub const COUNT: u8 = 2;
+}
+
+/// IDs for read-only allow buffers
+mod ro_allow {
+    /// Buffer for the message to be transmitted
+    pub const MESSAGE_WRITE: usize = 0;
+
+    /// Number of read-only allow buffers
+    pub const COUNT: u8 = 1;
+}
+
+/// IDs for read-write allow buffers
+mod rw_allow {
+    /// Buffer for the message to be received
+    pub const MESSAGE_READ: usize = 0;
+
+    /// Number of read-write allow buffers
+    pub const COUNT: u8 = 1;
+}
+
+enum OpType {
+    Tx,
+    Rx,
+}
+
+struct OpContext {
+    msg_tag: u8,
+    peer_eid: u8,
+    msg_type: u8,
+    op_type: OpType,
+}
+
+impl OpContext {
+    fn for_me(&self, msg_tag: u8, peer_eid: u8, msg_type: u8) -> bool {
```

Comment:
`for_me` is a little confusing. Maybe `matches` or `is_compatible`?

Reply:
Haha, I've been switching between `for_me` and `is_matched` before pushing the PR. Thanks for the help with the naming! I'll go with `matches`.